q2_moshpit/partition/mags.py

```diff
@@ -0,0 +1,142 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2023, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+import os
+import shutil
+import numpy as np
+from qiime2.util import duplicate
+from .utils import _validate_mag_ids, _validate_num_partitions
+from q2_types_genomics.per_sample_data import (
+    MultiMAGSequencesDirFmt
+)
+from q2_types_genomics.feature_data import (
+    MAGSequencesDirFmt
+)
+
+
```

Comment:
Hey @Sann5, is there any reason to keep the two partitioning/collating methods separate? Since they work with different semantic types and formats, it should be possible to just have one method for both, sample data and feature data, no?

Reply:
Oh yes. In a nutshell, because the return type is not the same qiime2 errors saying that the action is not well specified.